addons/web_editor/static/src/js/editor/odoo-editor/test/spec/insertHTML.test.js

```diff
@@ -79,6 +79,33 @@ describe('insert HTML', () => {
                 contentAfter: '<p>contentunwrapped</p><div><i class="fa fa-circle-o-notch"></i></div><p>culprit</p><p>after[]</p>',
             });
         });
+        it('should remove the trailing single <br> when pasting', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>[]<br></p>',
+                stepFunction: async editor => {
+                    await editor.execCommand('insert', parseHTML('<p>abc<br></p>'));
+                },
+                contentAfter: '<p>abc[]</p>',
+            });
+        });
+        it('should not remove the trailing single <br> when pasting between text', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>ab[]cd</p>',
+                stepFunction: async editor => {
+                    await editor.execCommand('insert', parseHTML('<p>efg<br></p>'));
+                },
+                contentAfter: '<p>abefg<br>[]cd</p>',
```

Comment:
That's a strange behavior actually because `<p>efg<br></p>` doesn't have a visible line break (hence your previous test) so I'd expect `<p>abefg[]cd</p>`.

@sebgeelen what do you think?

Reply:
It's probably correct since it's the kind of html you will get in the clipboard when you copy this : 
```
abcd[efg
]
hij
```

Reply:
Really? But this test and the one above it can't both be true at the same time, can they? And the one above seems right to me.
For what it's worth, I just tested in this `textarea` and it's inconsistent:
Pasting 
![image](https://github.com/odoo/odoo/assets/24205914/82dcb812-cbe3-41ee-a992-4c8d7b810d9a)
from your comment in here never includes the line break, while 
![image](https://github.com/odoo/odoo/assets/24205914/432f772f-4798-432a-8c64-76e1249fa04b)
does.

Reply:
Ok, 

The silver lining here is that this :  ``<p>efg<br></p>`` could append (in the paste clipboard).
So IMO, the test result make sense.

Reply:
OK so then the previous test doesn't. My bad because it's me who suggested it. Sorry @deso-odoo :pray: 

I tested in a bare contenteditable and indeed copying from `<p>[abc<br>]def</p>` gives the html `<p>abc<br></p>` in the clipboard, which makes contenteditable behave inconsistently since pasting it at `<p>hel[]lo</p>` will give  `<p>helabc<br>[]lo</p>` (which shows a visible line break) but pasting at `<p>hello[]</p>` will give `<p>helloabc<br>[]</p>` (which doesn't show a visible break so that the cursor looks like it's attached to the `c`).

Conclusion: even though the `br` at the end of `<p>abc<br></p>` isn't visible, in the clipboard it signifies a visible line break, therefore pasting `<p>abc<br></p>` should always show a visible line break at the end.

In other words:
```js
before: '<p>[]<br></p>'
paste('<p>abc<br></p>') // pasted html ends with 1 visible line break
after: '<p>abc<br>[]<br></p>' // 1 visible line break at the end (selection can be after the last, it makes no difference)
```
```js
before: '<p>ab[]cd</p>'
paste('<p>efg<br></p>') // pasted html ends with 1 visible line break
after: '<p>abefg<br>[]cd</p>' // 1 visible line break in the middle
```
```js
before: '<p>[]<br></p>'
paste('<p>abc<br><br></p>')  // pasted html ends with 2 visible line breaks
after: '<p>abc<br><br>[]<br></p>' // 2 visible line breaks at the end (selection can be after the last, it makes no difference)
```

Reply:
@Zynton Then we might not need the third commit, it can be done with the first commit itself. I have made the changes. It should be good now. :slightly_smiling_face: